test/src/test/java/org/corfudb/runtime/collections/CorfuTableTest.java

```diff
@@ -245,4 +255,47 @@ public void snapshotInvariant() {
         final Stream<Map.Entry<Integer, Integer>> result = map.entryStream();
         result.forEach(e -> map.put(new Random().nextInt(), 0));
     }
+
+    @Test
+    public void PersistedCorfuTableTests() {
```

Comment:
This test is using checkpointer specific logic, so it should be moved to `CheckpointTest`. A unit test should be checking for more specific functionality -- for example -- calling `hasNext()` on a closed iterator behaves as expected.

Reply:
Its not checkpoint specific, although the checkpointer uses a CorfuTable that doesn't work.

Reply:
This code was pretty much copied from `CheckpointWriter::appendObjectState`. Can you please add proper unit tests? Both positive and negative tests. You can add them to `DiskBackedCorfuClientTest`.

Reply:
This test belongs to CheckpointTest. Also, added more tests.